### DISCUSSION TITLE
#[Modelable] generating error in firefox

### DISCUSSION THREAD

**Discussion by PHPieter** [livewire/livewire#9073] [7. Bugs (archived)]
### Livewire version

v3.5.16

### Laravel version

v11.31.0

### Which PHP version are you using?

PHP 8.2

### What is the problem?

I also raised this issue in the flux github since i am not 100% sure if its a livewire or Flux problem. But it think it is Livewire.

https://github.com/livewire/flux/issues/821

Using firefox 133.0
When using `wire:model.live` in the root component. The root component has a child component which is bound using wire.model.
When changing the value of the root input. We get an error: 
`DOMException: Node.removeChild: The node to be removed is not a child of this node`

It looks like this error only exist when using a form object.
Also when changing the input of the child component we get an error:
`Uncaught TypeError: object is null`

This error occurs when the child component has at least two input fields.

### Root component
```
<?php

namespace App\Livewire;

use App\Livewire\Forms\TestForm;
use Livewire\Component;

class Test extends Component
{
    public $live_updated_attribute;
    public $modelable_attribute = [];

    public $boatTypeOptions = ['option1' => 1, 'option2' => 2];

    public function render()
    {
        return view('livewire.test');
    }
}
```

### Root component template
```
<div>
    <flux:select class="mb-3" wire:model.live="live_updated_attribute" variant="listbox" >
        @foreach($boatTypeOptions as $option =>$val)
            <flux:option :value="$val">{{ $option }}</flux:option>
        @endforeach
    </flux:select>

    <livewire:address-section wire:model="modelable_attribute"/>
</div>
```

### Child component
```
<?php

namespace App\Livewire;

use App\EnumClasses\DutchProvinceEnum;
use App\Rules\ZipCodeRule;
use Livewire\Attributes\Modelable;
use Livewire\Attributes\Validate;
use Livewire\Component;

class AddressSection extends Component
{
    #[Modelable]
    public $address = [];

    public function render()
    {
        return view('livewire.address-section');
    }
}

```

### Child component template
```
<div>
    <flux:input type="text" wire:model.live.debounce="address.street" label="{{__('address.Street')}}" :placeholder="__('address.Street')" />
    <flux:input type="text" wire:model.live.debounce="address.house_number" :label="__('address.House number')" :placeholder="__('address.House number')" />

</div>

```

### Wirebox URL

https://wirebox.app/b/y83o2

### How do you expect it to work?

I would Expect the the property `modelable_attribute` would contain the values of all the input field i put in the child component. This is actualy what is happening but in Firefox it also generates an error. I also figured out i need to put wire:model.live on it for it to work.  

Sorry but wirebox does not save my template files, so i can not test it. When i revisit the URL the template files look different than what i have typed.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by lrljoe** [comment]
For starters: 
"form.live_updated_attribute", you don't have a "form"

"address.street" doesn't exist to bind to...

This isn't a "bug", it's down to how you've written your app.

**Comment by PHPieter** [comment]
Thanks for pointing out my typo for this example. I removed the "form.". 

I was under the assumption address.street would be ok, because address does exist and is an array. It even works in Chrome and in Firefox to. Its just that i get the error witch makes it a bit weard.

**Reply by lrljoe** [reply]
Typically, I'd counsel against binding against an array, it's not clean, and it's not a nice way to bind.

Instead, your "AddressSection" component should receive an AddressID (for example), and then populate a Form.

Otherwise, you may run into issues when things get deleted/aren't present etc.
